### ISSUE TITLE
Drills rework, produce waste rock

### ISSUE THREAD

**Issue by hortiSquash** [Anuken/Mindustry-Suggestions#3067]
### Describe the content or mechanics you are proposing.

Issues :
- Incinerators largely unused
- Water extractors extremely overused
- Water overused in the game, both for boost and craft
- Most designs having the blocks directly touching each over, too simple and compact, not enough transport / sort
- Drills / water extractors not too realistic

Idea :
Make drills produce a second ressource, as waste. It may be dirt, pebbles, stone, mud, whatever. Something you find in the ground, that cant be used for anything. When you dig, you mostly get rock, not ressources.
- [x] more realistic drills
- [x] more use of the incinerator (waste has to go somewhere)
- [x] transport and sort required

Obviously, keep the same ressource output, just with additional waste. instead of 11/s AB, you go 20 but with 9/s waste for instance (random numbers)
I mean, if you manage to find a way to use pebbles/dirt as an actual usefull ressource, may not be a bad idea. Need some to craft cultivators for instance? Or melt rock into slag? Reduce into sand? hehehe quite interesting for future content...

As for water extractors, i had two ideas :
- First one : they produce unfiltered unusable water, which you have to filter in a new filter building, taking unfiltered water and coal as input (because coal filters), and make clean water out of that. You can even call it liquid sorter if you want, for the meme
- Second version : they produce mud or whatever, the same ressource as drills, as a "solid" item output. Nerf having water extractors directly next to other buildings.
- [x] water extractors extremely overused

### Describe how you think this content will improve the game. If you're proposing new content, mention how it may add more gameplay options or how it will fill a new niche.

Way more use of the incinerator. Change the way water extractors and drills are used, in a more interesting way design wise. Very interesting if you ask me. Would change many designs and meta without changing any ratios. As mentionned above, this "dirt/rock" ressource may be reused for other stuff in the future and add more gameplay options.

Perhaps it would be needed to buff a little the production of drills and watex to counteract this nerf of requiring some sorting/filtering of output.

### Before making this issue, check the boxes below to confirm that you have acknowledged them.

- [X] I have checked the [Trello](https://trello.com/b/aE2tcUwF/mindustry-trello) to make sure my suggestion isn't planned or implemented in a development version.
- [X] I am familiar with all the content already in the game or have glanced at the wiki to make sure my suggestion doesn't exist in the game yet.
- [X] I have read [README.md](https://github.com/Anuken/Mindustry-Suggestions/blob/master/README.md) to make sure my idea is not listed under the "A few things you shouldn't suggest" category.

**Comment by itcannotbe** [comment]
The games doesn't aim to be realistic.

**Comment by MrDuck557** [comment]
Oh no
too much complexity
seriously what is the point other than making drills uncompact

**Comment by IIINSTINCT** [comment]
now I feel like "realistic" is just a dumb reason to change something

**Comment by hortiSquash** [comment]
> Oh no too much complexity seriously what is the point other than making drills uncompact
honestly adding a single sorter wont change anything

**Comment by EclipseTheOldOne** [comment]
> > Oh no too much complexity seriously what is the point other than making drills uncompact
> 
> honestly adding a single sorter wont change anything

Designer Fact: it does, and a lot

**Comment by hortiSquash** [comment]
![image](https://user-images.githubusercontent.com/45213805/135883835-cebedcb4-758d-4bc5-a9ac-8a66f0b7f62e.png)
we longly discussed about it in #design before i posted it, this screenshot is from #balancing

**Comment by OnyxFinger** [comment]
> Water extractors extremely overused

yes, but this is a mistake, it's not because something is overused that it's OP. Usually, if you can use water from other sources, it's simply better

> Water overused in the game, both for boost and craft

not for craft, but it is used a lot for boost. I don't see where that's an actually problem though

> Most designs having the blocks directly touching each over, too simple and compact, not enough transport / sort

more transport stuff would be absolutely pointless in cases other than making only a few schems eve so slightly more compact

> Drills / water extractors not too realistic

the game isn't about realism, in fact, I almost feel like realism sometimes takes over too hard

waste : ok but.. what's the point ?

> Way more use of the incinerator.

BUT. Noone likes the idea of having to sort everything, also what dya do when you don't have the ressources for them yet ?

**Comment by begginerr** [comment]
> > Water extractors extremely overused
> 
> yes, but this is a mistake, it's not because something is overused that it's OP. Usually, if you can use water from other sources, it's simply better
> 
> > Water overused in the game, both for boost and craft
> 
> not for craft, but it is used a lot for boost. I don't see where that's an actually problem though
> 
> > Most designs having the blocks directly touching each over, too simple and compact, not enough transport / sort
> 
> more transport stuff would be absolutely pointless in cases other than making only a few schems eve so slightly more compact
> 
> > Drills / water extractors not too realistic
> 
> the game isn't about realism, in fact, I almost feel like realism sometimes takes over too hard
> 
> waste : ok but.. what's the point ?
> 
> > Way more use of the incinerator.
> 
> BUT. Noone likes the idea of having to sort everything, also what dya do when you don't have the ressources for them yet ?

he is very right

**Comment by EclipseTheOldOne** [comment]
![Screenshot_414](https://user-images.githubusercontent.com/43229798/135942630-62624b76-2f7c-41ed-b15e-5fac95fafd03.png)
well.

**Comment by the-red-soul** [comment]
>>Make drills produce a second ressource, as waste. It may be dirt, pebbles, stone, mud, whatever. Something you find in the ground, that cant be used for anything. When you dig, you mostly get rock, not ressources

if it has literally no function then there is absolutely no reason to include this in the game. It is bloat for the sake of bloat and inconvenience. 

By your logic of drills producing waste because they are inefficient then i guess every single factory will have to also produce waste since no irl factory is 100% efficient with zero waste. Having different kinds of waste for every single factory. Totally not a waste of time to implement and sprite every single waste item when it just gets burned.

>>Most designs having the blocks directly touching each over, too simple and compact, not enough transport / sort

Why is this a bad thing? its how the game works. All factories/drills/extractors output to what is next to them no matter if it is a conveyor, conduit, or other factory. It is how the game was designed in the first place for ease of use and simplicity when programming.

>>Incinerators largely unused

This is a problem why? Spore is unused do you want to add some useless function to it as well? What about scorch turret? what about containers instead of vaults? A LOT of things in this game are underused compared to others. This is not really an issue since they all have their use cases, unlike having unusable waste.

>>Water extractors extremely overused
>>First one : they produce unfiltered unusable water, which you have to filter in a new filter building, taking unfiltered water and coal as input (because coal filters), and make clean water out of that. You can even call it liquid sorter if you want, for the meme

 A pointless liquid with one extra step to be useable... Why would it be tainted water anyway? That does not make sense in the name of Water Extractor, and for all you know filters are prebuilt into the building. Would your coal "Brita" water filters have waste product as the coal is used up? More pointless bloat.

Second version : they produce mud or whatever, the same ressource as drills, as a "solid" item output. Nerf having water extractors directly next to other buildings.

This would not even do anything since items do not go INTO drills, and factories do not accept the wrong kind of items. *Also still pointless bloat*

If you want to nerf water extractor you nerf the amount of water it gives. I believe Anuke at one point was trying to do some proximity water/oil extractor things... Dont know how it went however.

>>Drills / water extractors not too realistic

neither is having infinite ores. neither is having fixed enemy spawn zones. neither is having cryo tiles in maps. neither is mega being able to carry a bryde since its bigger than mega. neither is spider units not taking damage on their legs. there are a LOT of things that are *not realistic.* IT IS A GAME. ITS NOT SUPPOST TO BE.

**Comment by acoaco** [comment]
bind flare throw out mud,problem solved lol

water is hard to implemented cuz you can just let it leak,though I think mud is hard too and meaningless just a sorter+incinerator around every airblast,is it fun? you just let single airblast drill become meaningless people will use airblast drill+sorter+incinerator+a pipe leak polluted water as default airblast set up lol

basically you need glass before get water now,water ex already got nerf
if it's near water people will build pump instead brainless spawn water ex to save power and resources now,that enough

I don't know why some specific people always keep want nerf water ex or airblast drill,it's not first time

realistic?why don't you said mend projector is unreal,it's should not exist building will become old and collapsed by time
why unit doesn't need any fuel but can keep move it's impossible for a little flare has perpetual motion

because you only want target airblast drill and water ex want nerf them not for realistic

though it's doesn't affect me(logic mining),the more nerf on basic block,logic become stronger and airblast drill+water ex become a trash,every one just watch a pure 200-300 pulsar/mega mining instead build drill

**Comment by Kyllingene** [comment]
@hortiSquash, the problem isn't that the game you want Mindustry to be is a bad game; the problem is that not only is it not trying to be that game, there is already such a game: Factorio. It is also a tower-defense base-building resource-management game, but much more complex.
